util-hadoop/src/test/java/com/google/cloud/hadoop/util/RefreshTokenAuth2ProviderTest.java

```diff
@@ -0,0 +1,160 @@
+package com.google.cloud.hadoop.util;
+
+import static com.google.common.truth.Truth.assertThat;
+import static net.jadler.Jadler.initJadlerUsing;
+import static net.jadler.Jadler.onRequest;
+import static net.jadler.Jadler.port;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import junit.framework.TestCase;
+import net.jadler.Jadler;
+import net.jadler.stubbing.server.jdk.JdkStubHttpServer;
+import org.apache.hadoop.shaded.net.minidev.json.JSONObject;
+import org.apache.http.HttpHeaders;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RefreshTokenAuth2ProviderTest extends TestCase {
+
+  @Before
+  public void before() {
+    initJadlerUsing(new JdkStubHttpServer());
+  }
+
+  @After
+  public void after() {
+    Jadler.closeJadler();
+  }
+
+  @Test
+  public void testGetAccessTokenWithoutProxy() throws IOException {
```

Comment:
For consistency, could we use `MockHttpTransport` from `com.google.api.client.testing.http` instead of `jalder` ?

Reply:
Sure.

Have you got an example that does roughly the same thing that I just did here?

Note that we are testing as well a proxy, MockHttpTransport would need to support that.

Reply:
May be try [Helper class](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/20f44b15b2a05adde6080783cc7772c1df22445d/util/src/main/java/com/google/cloud/hadoop/util/testing/MockHttpTransportHelper.java#L52) , refer to usages in [test](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannelTest.java#L76)

Reply:
Jadler was a better testing solution, as it actually tested the request body and even http call, although I understand that for consistency, you prefer to use the MockHttpTransport instead.
Note: MockHttpTransport is bypassing the proxy configuration which therefore can't be tested.